schemas/EGA.submission.json

```diff
@@ -4,10 +4,188 @@
+            "properties": {
+              "ega_accession": {
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/EGA-submission-id-pattern"
+              }
+            }
+          }
+        ]        
+      },
+
+      "schema_descriptor": {
+        "title": "Schema descriptor node",
+        "description": "Inherited schema descriptor node containing metadata about the schemas and standards used to create the JSON document itself.",
+        "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/schema_descriptor"
+      },
+
+      "object_title": {
+        "type": "string",
+        "title": "Title of the submission project",
+        "description": "Short free-form text that can be used to call out submission project records in searches or displays.",
+        "minLength": 1,
+        "examples": [ "Submission XF40" ]
+      },
+
+      "object_description": {
+        "type": "string",
+        "title": "Description of the submissions project",
+        "description": "An in-depth description of the submission, including its overall purpose or nature of studies it governs.",
+        "minLength": 1,
+        "examples": [ "EBI submission project XF40 of 1000 samples and its 1500 sequencing runs" ]
+      },
+
+      "additional_collaborators": {
+        "type": "array",
+        "title": "Submission ownership details",
+        "description": "Object containing optional collaborators of the submission project, who shall have different capabilities granted by the owner: 'read' or 'read and write' rights.",
+        "minItems": 1,
+        "additionalProperties": false,
+        "items": { 
+          "type": "object",
+          "title": "Collaborator",
+          "description": "Collaborator item comprising both the collaborator's contact details and rights.",
+          "additionalProperties": false,
+          "required": ["collaborator_rights", "collaborator_contact_details"],      
+          "properties": {
+            "collaborator_rights": {
+              "type": "string",
+              "title": "Collaborator rights",
+              "description": "Property defining the rights of the specified collaborator. Either read-only or read and write rights.",
+              "enum": [ "read_only", "read_and_write" ],
+              "meta:enum": {
+                "read_only": "Collaborator will only be capable of reading the details of the submission.",
+                "read_and_write": "Collaborator will be able to not only read, but modify or add changes to the submission."
+              }
+            },
+            "collaborator_contact_details": {
+              "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/contact_details"
+            }
+          }
+        }
+      },
```

Comment:
How does this information align with the EBI's current submission process, whereby a submission account (i.e. ega-box-###) is opened for all named individuals in the submission statement? How does this information align with CRG's proposed approach for managing submission accounts?

I'm assuming the information provided by a submitter here doesn't have any actual effect on who can read from / write to a submission account. Could this be confusing to users?

Reply:
That is a really good question. The conflict was to have a JSON Schema we can share metadata between institutions with, but also a JSON Schema that users could use to provide said metadata. 

Based on the demos we saw of the Submitter Portal, users can share their submissions and collaborate. CRG members also agreed to have this information within the JSON documents. **Although we did not mention explicitly that what users put into this node would govern who can collaborate with which submission, that was what I understood**. 

If, instead, it's preferable to have this information provided in another way, we could remove it from the submission JSON schema. We can also create an extended JSON schema with that additional node for EBI-CRG use only. In any case, I would advise to keep this node and also effectively reading collaborators from it.

Regarding how it aligns with EBI's / CRG: the way we created this object was **mainly having CRG's point of view of a submission**. Since these schemas are expected to be use in the long-run, and we will eventually reproduce/reuse this submission concept, I disregarded the current submission concept (perhaps ADD/MODIFY actions could come in handy, though, but that would be for the object-set, not for this concept of submission). I imagine that, either at some point we adapt submission accounts at EBI in a similar fashion to CRG (after all we'll be using the same website), or we can have collaborators being added individually by their email addresses, apart from submission accounts.

Let me know if that answers your question or not, happy to discuss if needed.

Reply:
It's a good idea to encode this information somewhere, and the submission JSON schema seems a good choice. What I don't think we can do is allow permissions to be set by someone providing a submission JSON document. This approach is effectively letting anyone who has access to a submission also control who else can access the submission (which might contain private info) without authorisation. It is only the Data Controller / DAC who can authorise collaborators, and this all has to be done via the DAC Portal or via Helpdesk.

I'm assuming because the name of the attribute is `additional_collaborators` that the data controllers and/or DACs are **not** meant to be listed in this block? 

tl;dr: The `additional_collaborators` block can be used to represent additional (or all?) collaborators, but changes to collaborator information must always go through the DAC Portal or Helpdesk.

Reply:
That's true, and I started to think on whether this may be too confusing for the users: having the node there but not being able to use it. 

Perhaps "submissions" could be submitted through another pathway (i.e. through DAC Portal). I'm not familiar with the DAC Portal, but as far as I remember it has no programmatic interaction, does it? I may be wrong, but during the Submitter Portal demo I think they showed how collaborators could be added directly without going through the DAC Portal. I need to check the recording.

In any case, we could always display in bold that the collaboration property is just for internal usage, and that it won't propagate if it's submitted directly. 

> I'm assuming because the name of the attribute is additional_collaborators that the data controllers and/or DACs are not meant to be listed in this block?

Having the real "owners" of the "submission" there was an initial approach, but I removed them, since the owners of an object are omitted in the other objects. But perhaps these details would come in handy when institutions exchange information. I wouldn't mix too much what users can submit and what institutions exchange as extras, since it leads to, for instance, the current dilemma of collaborators.